atom/browser/extensions/atom_component_extension_resource_manager.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright 2014 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "atom/browser/extensions/atom_component_extension_resource_manager.h"
+
+#include "base/logging.h"
+#include "base/macros.h"
+#include "base/path_service.h"
+#include "build/build_config.h"
+#include "brave/grit/brave_resources.h"
+#include "chrome/common/chrome_paths.h"
+#include "chrome/grit/component_extension_resources_map.h"
+
+namespace extensions {
+
+AtomComponentExtensionResourceManager::
+AtomComponentExtensionResourceManager() {
+  static const GritResourceMap kExtraComponentExtensionResources[] = {
+    {"cryptotoken/manifest.json", IDR_CRYPTOTOKEN_MANIFEST},
+  };
+
+  AddComponentResourceEntries(
+      kComponentExtensionResources,
+      kComponentExtensionResourcesSize);
+  AddComponentResourceEntries(
+      kExtraComponentExtensionResources,
+      arraysize(kExtraComponentExtensionResources));
+}
+
+AtomComponentExtensionResourceManager::
+~AtomComponentExtensionResourceManager() {}
+
+bool AtomComponentExtensionResourceManager::IsComponentExtensionResource(
```

Comment:
most of this code looks identical to `ChromeComponentExtensionResourceManager`. We want to avoid copying chrome code so can you subclass instead?